app/controlplane/internal/usercontext/currentuser_middleware.go

```diff
@@ -110,8 +111,44 @@ func WithCurrentUserAndOrgMiddleware(userUseCase biz.UserOrgFinder, logger *log.
 	}
 }
 
-// Find organization and user in DB
-func setCurrentOrgAndUser(ctx context.Context, userUC biz.UserOrgFinder, userID string, logger *log.Helper) (context.Context, error) {
+func WithCurrentOrganizationMiddleware(userUseCase biz.UserOrgFinder, logger *log.Helper) middleware.Middleware {
+	return func(handler middleware.Handler) middleware.Handler {
+		return func(ctx context.Context, req interface{}) (interface{}, error) {
+			rawClaims, ok := jwtMiddleware.FromContext(ctx)
```

Comment:
Actually it's need to prevent requests generated from an API token to execute this logic since they will fail because they don't have a User.

The API token is tried to configured on the first middleware of the chain.

Reply:
but wouldn't be the same if you just check if the user is set?

```
u := CurrentUser(ctx)
if u == nil {
		handler(ctx, req)
	}
```

Reply:
Yes, that would also work, thanks.